src/Cardano/Wallet/Api/Types.hs

```diff
@@ -202,91 +199,111 @@ getApiMnemonicT (ApiMnemonicT (pw, _)) = pw
                              Encoding & Decoding
 -------------------------------------------------------------------------------}
 
-newtype DecodeApiAddressError = DecodeApiAddressError String
-    deriving stock Show
-    deriving newtype Buildable
+-- | Defines a textual encoding for a type defined within the API.
+class ToText a where
+    -- | Encode the specified value as text.
+    toText :: a -> Text
 
--- | Constructs an address from a Base58-encoded string.
---
--- Fails if the specified string is not Base58 encoded.
-decodeApiAddress :: Text -> Either DecodeApiAddressError (ApiT Address)
-decodeApiAddress x = maybe
-    (Left $ DecodeApiAddressError
-        "Unable to decode Address: expected Base58 encoding")
-    (pure . ApiT . Address)
-    (decodeBase58 bitcoinAlphabet $ T.encodeUtf8 x)
-
--- | Converts an address to a Base58-encoded string.
-encodeApiAddress :: ApiT Address -> Text
-encodeApiAddress =
-    T.decodeUtf8 . encodeBase58 bitcoinAlphabet . getAddress . getApiT
-
-newtype DecodeApiEncryptionPassphraseError
-    = DecodeApiEncryptionPassphraseError String
+-- | Defines a textual decoding for a type defined within the API.
+class FromText a where
+    -- | Decode the specified text as a value.
+    fromText :: Text -> Either TextDecodingError a
+
+-- | Indicates an error that occurred while decoding from text.
+newtype TextDecodingError = TextDecodingError
+    { getTextDecodingError :: String }
```

Comment:
Not sure what's the value of the `newtype` here. I'd simply go for a string to be honest. It adds value when we have multiple ones, one per actual type (albeit having a raw string isn't really the best option). But since it has now become a common abstraction, a `String` has pretty much the same value as the newtype here.

Reply:
I think there is some value in having a `newtype` here.

1. The descriptive name of the type `TextDecodingError` is self-commenting. When I see this type in the wild somewhere, I don't need to think very hard to know that it's an error relating to the decoding of text, and not some random string that happens to have been inserted into the left-hand-side of an `Either`.

2. If I see `TextDecodingError` in the wild, in a place _outside_ of the `Api.Types` module, then I have a direct way to look up its definition, at which point I can see the extended comment.

3. If I were to try refactor `TextDecodingError` in some way, then I might get better compiler feedback than if it were just a naked `String`.

What do you think? Of course, if you think the arguments I've presented have no basis, then we can remove it.

(Note that I've now created a separate PR for the textual encoding work: #142.)